### PR TITLE
Create a new ad slot PoC for driving traffic to GLabs campaigns.

### PR DESCRIPTION
**Pull request by JonNorman**
## What does this change?
This is a new ad slot, which will sit at the top of articles, and will be intended to drive traffic to GLabs content.

This slot sits behind a 0% ad test such that it can be opted into while we try out new designs.

@guardian/commercial-dev

### CODE REVIEW

**Comment by JonNorman** [comment]
[Add Carrot Slot behind 0% ad test](https://trello.com/c/4iI7RPRG/138-add-carrot-slot-behind-0-ad-test)

**Comment by regiskuckaertz** [comment]
:carrot: 🐰

File: static/src/javascripts/projects/commercial/modules/carrot-slot.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const carrotSlotInit = () => {
+    if (commercialFeatures.carrotSlot) {
+        const anchorSelector = '.js-carrot';
+        const anchor = document.querySelector(anchorSelector);
+
+        const slot = createSlot('carrot');
+
+        return fastdom
+            .write(() => {
+                if (anchor && anchor.parentNode) {
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/carrot-slot.js]: just `anchor.insertAdjacentElement('beforebegin', slot)` will do

**Comment by JonNorman** [comment]
Etymology of the name "carrot" was my eating carrots when I had to come up with a name for the style. But it has since become apparent that the slot in question is narrow and long and intended to entice people to click on it, such is the humble carrot.

File: static/src/javascripts/projects/commercial/modules/dfp/create-slot.js [hunk 2]
```diff
@@ -62,6 +62,14 @@ const adSlotDefinitions = {
             ],
         },
     },
+    carrot: {
+        label: false,
+        refresh: false,
+        name: 'carrot',
+        sizeMappings: {
+            mobile: [adSizes.fluid],
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/dfp/create-slot.js]: YEAH! First slot 100% fluid 🎉 ❤️

File: static/src/stylesheets/module/_adslot.scss [hunk 3]
```diff
@@ -414,6 +414,11 @@
     padding: 0;
     margin: 0;
 
+    &.ad-slot--carrot {
```
Comment (katebee) [inline static/src/stylesheets/module/_adslot.scss]: Does this need to be nested? Could we move it to after `ad-slot--fluid`? That way we keep the scss flatter, and can remove the ampersand:

```
    .... 
} // end of ad-slot--fluid

.ad-slot--carrot {
    min-height: 0;
    margin-bottom: 16px;
}

.ad-slot--fabric-v1 {
    min-height: 250px;
}
```

**Comment by LATaylor-guardian** [comment]
@JonNorman As I mentioned - it might be worth creating some lightweight documentation about how to do this (linking to this PR) as it seems like the kind of thing someone on this team may have to do again.

**Comment by JonNorman** [comment]
@LATaylor-guardian Yep sounds good, I'll think of a good place to put it.

File: static/src/stylesheets/module/_adslot.scss (hunk 3 above)
Comment (JonNorman) [inline static/src/stylesheets/module/_adslot.scss]: Ah yeah there's no need for the nesting. Thanks!

**Review by katebee — APPROVED** [review]
👍 💎 Fantastic, can't wait to see it!

Might be worth giving it a feature switch... but that can probably wait until we complete the AB test?

**Comment by JonNorman** [comment]
@katebee Good point, I'll just add it in now, since it'll only take a second :)

**Comment by JonNorman** [comment]
Some documentation for wider context: https://docs.google.com/a/guardian.co.uk/document/d/1YA5lp8Pm6ZqhJ_Z76mPImqT1A_6TbGS73XNGwcVYOmw/edit?usp=sharing

**Pull request merged**